### PR TITLE
feat(GaussDB): add gaussdb mysql database resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add gaussdb mysql database resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add gaussdb mysql database resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb/' TESTARGS='-run TestAccGaussDBDatabase_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb/ -v -run TestAccGaussDBDatabase_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussDBDatabase_basic
=== PAUSE TestAccGaussDBDatabase_basic
=== CONT  TestAccGaussDBDatabase_basic
--- PASS: TestAccGaussDBDatabase_basic (839.61s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   839.700s
```

### CODE REVIEW

File: docs/resources/gaussdb_mysql_database.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  Changing this parameter will create a new resource.
+
+* `description` - (Optional, String) Specifies the database remarks. The value can consist of up to 512 characters,
+  and cannot contain the carriage return characters or special characters (!<"='>&).
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
```
Comment (Jason-Zhang9309) [inline docs/resources/gaussdb_mysql_database.md]:
```suggestion
* `id` - The resource ID. The format is <instance_id>/<name>.
```

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_database.go [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+}
+
+func buildGaussDBMysqlQueryParams(offset int) string {
+	return fmt.Sprintf("?limit=100&offset=%v", offset)
+}
+
+func updatePathOffset(path string, offset int) string {
+	index := strings.Index(path, "offset")
+	return fmt.Sprintf("%soffset=%v", path[:index], offset)
+}
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_database.go]: looks this func can be replaced by buildGaussDBMysqlQueryParams

File: huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_mysql_database_test.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+}
+
+func buildGaussDBMysqlQueryParams(offset int) string {
+	return fmt.Sprintf("?limit=100&offset=%v", offset)
+}
+
+func updatePathOffset(path string, offset int) string {
+	index := strings.Index(path, "offset")
+	return fmt.Sprintf("%soffset=%v", path[:index], offset)
+}
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_mysql_database_test.go]: use buildGaussDBMysqlQueryParams?

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/gaussdb/' TESTARGS='-run TestAccGaussDBDatabase_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb/ -v -run TestAccGaussDBDatabase_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussDBDatabase_basic
=== PAUSE TestAccGaussDBDatabase_basic
=== CONT  TestAccGaussDBDatabase_basic

--- PASS: TestAccGaussDBDatabase_basic (871.98s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   900.611s

**Pull request merged**